SQLite/Core/Connection.swift

```diff
@@ -25,8 +25,209 @@
 import Dispatch
 import CSQLite
 
+
+/// The mode in which a transaction acquires a lock.
+public enum TransactionMode : String {
+    
+    /// Defers locking the database till the first read/write executes.
+    case Deferred = "DEFERRED"
+    
+    /// Immediately acquires a reserved lock on the database.
+    case Immediate = "IMMEDIATE"
+    
+    /// Immediately acquires an exclusive lock on all databases.
+    case Exclusive = "EXCLUSIVE"
+    
+}
+
+
+/// Protocol to an SQLite connection
+public protocol Connection {
```

Comment:
Also, I kind of liked having `DBConnection`, regardless of the protocol name, simply because it has more capabilities than the protocol does and the name kind of explains why in that case.  Although, with a renamed protocol, simply `Connection` works as well.